cypress/integration/channel/add_users_to_channel_spec.js

```diff
@@ -0,0 +1,102 @@
+        cy.apiLogin('user-1');
+        cy.visit('/');
+
+        // # Select 'doloremque' channel on the left hand side
+        cy.get('#sidebarItem_saepe-5').click();
+
+        // * 'doloremque' shoul be visible at the top of the center pane
+        cy.get('#channelHeaderTitle').should('contain', 'doloremque');
+
+        // # Then click it to access the drop-down menu
+        cy.get('#channelHeaderTitle').click();
+
+        // * The dropdown menu of te channel header should be visible;
+        cy.get('#channelHeaderDropdownMenu').should('be.visible');
+
+        // # Click 'Add Members'
+        cy.get('#channelAddMembers').click();
+
+        // * Assert that modal appears
+        cy.get('#addUsersToChannelModal').should('be.visible');
+
+        // # Wait 1 second
+        cy.wait(1000);// eslint-disable-line
+
+        // # Click the first row clickable of the modal to select a user
+        cy.get('#userAddList').first().click();
+
+        // # Click the button "Add" to add this user to 'doloremque' channel
+        cy.get('#saveItems').click();
+
+        cy.getLastPostId().then((id) => {
+            // # On the system message advising a user was added to the channel, click the username link
+            cy.get(`#post_${id}`).find('.mention-link').first().click({force: true}).then(($el) => {
+                // * Then the user's profile pop-over should be displayed when their username is clicked, and the pop-over is fully visible
+                cy.get('#user-profile-popover').should('be.visible');
+
+                // * The pop-over title  at the top of the pop-over should be the same as the username link
+                const userName = $el[0].innerHTML;
+                const escapedUsername = userName.replace('@', '').replace('.', '-');
+                cy.get(`#userPopoverUserName-${escapedUsername}`).should('contain', userName);
+            });
+        });
+
+        // # Select 'sint', another channel on the left hand side
+        cy.get('#sidebarItem_suscipit-4').click();
+
+        // # Click 'sint' at the top of the center pane to access the drop-down menu
+        cy.get('#channelHeaderTitle').click();
+
+        // * The dropdown menu of the 'sint' channel header should be visible;
+        cy.get('#channelHeaderDropdownMenu').should('be.visible');
+
+        // # Click 'Add Members'
+        cy.get('#channelAddMembers').click({});
+
+        // * Assert that modal appears
+        cy.get('#addUsersToChannelModal').should('be.visible');
+
+        // # Click 3 times the first row clickable of the modal to select 3 different users
+        cy.get('#userAddList').first().click();
+        cy.get('#userAddList').first().click();
+        cy.get('#userAddList').first().click();
+
+        // # Click the button "Add" to add these 3 users to 'sint' channel
+        cy.get('#saveItems').click();
+
+        // # Wait 2 seconds to allow the popup to disappear
+        cy.wait(2000); // eslint-disable-line
```

Comment:
`cy.get('#foobarbaz').should('be.hidden')` or cy.get('#foobarbaz').should('not.exist')` (depending on what happens to it) for the appropriate element instead of waiting.

Reply:
Hey @thekiiingbob , I tried but it didn't work so I used TIMEOUT instead

Reply:
@VolatianaYuliana `cy.get('#addUsersToChannelModal').should('not.exist');` for the 2 places you were waiting for the modal to go away, worked for me. Was there a specific error/issue you were running into?

Reply:
This is what I got when I use cy.get('#addUsersToChannelModal').should('not.exist') instead of waiting... but it works when I use TIMEOUT.
![error](https://user-images.githubusercontent.com/45851538/58885009-18ea7a80-86ea-11e9-8517-a58c2fa5de6e.png)

Reply:
@VolatianaYuliana Looks like the wait for the element to not exists works properly, but the post isn't ready yet? Locally, even with induced network latency, I cannot reproduce this error. Is the post still loading for you? Seems weird because the check for the post to have and id and also not be pending looks to be passing for you so it SHOULD be ready.

Reply:
The 'added to the channel by you' text is not yet visible in the last post even if the check for the last post to have id is passing  ...as you can see in the screenshot, 
![error2](https://user-images.githubusercontent.com/45851538/58895000-abdfe080-86fb-11e9-9151-f8f5621c2dc4.png)

Reply:
it's weird why it doesn't work for me

Reply:
Mind if I get the specs of your development machine? It feels like it's a timing issue of the system not posting the new message for you as quick as we want it to. Ideally we don't want to do a hard wait, so let me think about what we can wait for.

Reply:
Here are my machine specs: 
![specs](https://user-images.githubusercontent.com/45851538/58898621-9ff81c80-8703-11e9-9d88-aee6d217ea6a.png)